corehq/apps/data_interfaces/templates/data_interfaces/explore_case_data.html

```diff
@@ -40,10 +46,104 @@ <h1>{% trans "Searching the Case Lists" %}</h1>
     </div>
   </div>
 
-  <table id="report-datatable" class="table"></table>
+  <div id="report-datagrid">
+    <table class="datagrid-headers">
+      <thead>
+        <tr>
+          <!-- ko foreach: columns -->
+          <th data-bind="style: { width: width() + 'px' }">
+            <span data-bind="text: title"></span>
+            <button type="button"
+                    class="btn btn-default"
+                    data-bind="click: $root.editColumnController.set">
+              {% trans "Edit" %}
+            </button>
+          </th>
+          <!-- /ko -->
+          <th>
+            <button class="btn btn-default"
+                    type="button"
+                    data-bind="click: $root.editColumnController.setNew">
+              {% trans "Add Column" %}
+            </button>
+          </th>
+        </tr>
+      </thead>
+    </table>
+    <table class="datagrid-rows">
+      <tbody data-bind="foreach: { data: data.rows, as: 'row' }">
+        <tr>
+          <!-- ko foreach: { data: $root.columns, as: 'column' } -->
+            <td data-bind="style: { width: column.width() + 'px' }">
+              <span data-bind="text: row[column.slug()]"></span>
+            </td>
+          <!-- /ko -->
+          <td></td>
+        </tr>
+      </tbody>
+    </table>
 
-</div>
-{% endblock %}
+    <div class="modal fade"
+         data-bind="modal: editColumnController.column"
+         role="dialog">
+      <div data-bind="with: editColumnController.column"
+           class="modal-dialog">
+        <div class="modal-content">
+          <div class="modal-header">
+            <button type="button" class="close" data-dismiss="modal">
+                <span aria-hidden="true">&times;</span>
+                <span class="sr-only">{% trans "Close" %}</span>
+            </button>
+            <h4 class="modal-title" data-bind="visible: $root.editColumnController.isNew">
+              {% trans "Add Column" %}
+            </h4>
+            <h4 class="modal-title" data-bind="visible: !$root.editColumnController.isNew()">
+              {% trans "Edit Column" %}
+            </h4>
+          </div>
+          <div class="modal-body">
+            <div class="form">
+              <div class="form-group">
+                <input type="text"
+                       class="form-control"
+                       placeholder="{% trans "Title" %}"
```

Comment:
Nitpick: `trans_html_attr` instead of `trans` will handle any escaping issues with the translation.

Reply:
oh thank you!